File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> anagramGroups;
        for (string s : strs) {
        string sortedStr = s;         
        sort(sortedStr.begin(), sortedStr.end());  
        
        
        anagramGroups[sortedStr].push_back(s);
    }
    
    
    vector<vector<string>> result;
    for (auto& group : anagramGroups) {
        result.push_back(group.second);
    }
    
    return result; 
    }
};

File: 49-group-anagrams/group-anagrams.py
from collections import defaultdict

class Solution:
    def groupAnagrams(self, strs: list[str]) -> list[list[str]]:
        res = defaultdict(list)
        
        for s in strs:
            alpha = [0] * 26
            for c in s:
                alpha[ord(c) - ord('a')] += 1
            res[tuple(alpha)].append(s)

        return list(res.values())